dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/dataset/deaccession/DatasetDeaccessionBean.java

```diff
@@ -0,0 +1,49 @@
+package edu.harvard.iq.dataverse.dataset.deaccession;
+
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion.VersionState;
+import edu.harvard.iq.dataverse.search.IndexServiceBean;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+@Stateless
+public class DatasetDeaccessionBean {
```

Comment:
TBH I think that functionality enclosed in this class should be moved to DatasetDeaccessionService

Reply:
Good point – I put it into this class in order to be able to access it as a container-proxied method, but now I see that in this case it's not really needed.